tests/msc2716_test.go

```diff
@@ -1342,36 +1383,12 @@ func validateBatchSendRes(t *testing.T, c *client.CSAPI, roomID string, batchSen
 	expectedEventIDOrder = append(expectedEventIDOrder, insertionEventID)
 
 	if validateState {
-		// Get a pagination token for the newest-in-time event in the historical batch itself
 		contextRes := c.MustDoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "context", expectedEventIDOrder[0]}, client.WithContentType("application/json"), client.WithQueries(url.Values{
 			"limit": []string{"0"},
 		}))
-		contextResBody := client.ParseJSON(t, contextRes)
-		batchStartPaginationToken := client.GetJSONFieldStr(t, contextResBody, "end")
-
-		// Fetch a chunk of `/messages` which only contains the historical batch. We
-		// want to do this because `/messages` only returns the state for the first
-		// message in the `chunk` and we want to be able assert that the historical
-		// state is able to be resolved.
-		messagesRes := c.MustDoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "messages"}, client.WithContentType("application/json"), client.WithQueries(url.Values{
-			// Go backwards (newest -> oldest) (same direction as if you were using scrollback)
-			"dir": []string{"b"},
-			// From the newest-in-time event in the historical batch
-			"from": []string{batchStartPaginationToken},
-			// We are aiming to scrollback to the oldest-in-time event from the
-			// historical batch
-			"limit": []string{fmt.Sprintf("%d", len(expectedEventIDOrder))},
-			// We add these options to the filter so we get member events in the state field
-			"filter": []string{"{\"lazy_load_members\":true,\"include_redundant_members\":true}"},
```

Comment:
`/messages` was only returning member state events and only because we passed this special filter. Now that we're testing with non-member state events, we need an endpoint which will show us them.

We can instead simplify and just use `/context` directly 🎉

Reply:
Bear in mind that `/context` does not work over federation. That is, if server A is in room X and sends 100 messages, then server B joins room X and hits `/context` for an event early in the timeline (so not backfilled) it will 404.

Reply:
I moved the `validateState` block (which calls `/context`) after we call `/messages` which will backfill any of the potential gaps 👍